repos/system_upgrade/el7toel8/actors/ziplcheckbootentries/libraries/ziplcheckbootentries.py

```diff
@@ -0,0 +1,134 @@
+from collections import defaultdict
+
+from leapp import reporting
+from leapp.libraries.common.config import architecture
+
+FMT_LIST_SEPARATOR = '\n    - '
+ZIPL_CONFIG_PATH = '/etc/zipl.conf'
+
+
+def is_rescue_entry(boot_entry):
+    """
+    Determines whether the given boot entry is rescue.
+
+    :param BootEntry boot_entry: Boot entry to assess
+    :return: True is the entry is rescue
+    :rtype:  bool
+    """
+    return 'rescue' in boot_entry.kernel_image.lower()
+
+
+def inhibit_if_multiple_zipl_rescue_entries_present(bootloader_config):
+    """
+    Inhibits the upgrade if we are running on s390x and the bootloader configuration
+    contains multiple rescue boot entries.
+
+    A boot entry is recognized as a rescue entry when its title contains the `rescue` substring.
+
+    :param SourceBootloaderConfiguration bootloader_config: The configuration of the source boot loader.
+    """
+
+    # Keep the whole information about boot entries not just their count as
+    # we want to provide user with the details
+    rescue_entries = []
+    for boot_entry in bootloader_config.entries:
+        if is_rescue_entry(boot_entry):
+            rescue_entries.append(boot_entry)
+
+    if len(rescue_entries) > 1:
+        # Prepare the list of available rescue entries for user
+        rescue_entries_text = ''
+        for rescue_entry in rescue_entries:
+            rescue_entries_text += '{0}{1}'.format(FMT_LIST_SEPARATOR, rescue_entry.title)
+
+        reporting.create_report([
+            reporting.Title('Multiple rescue boot entries present in the bootloader configuration.'),
+            reporting.Summary(
+                'The zipl configuration file {0} contains multiple rescue boot entries:{1}'
+                .format(ZIPL_CONFIG_PATH, rescue_entries_text)
+            ),
+            reporting.Severity(reporting.Severity.HIGH),
+            reporting.Tags([reporting.Tags.BOOT]),
+            reporting.Remediation(hint='Remove rescue boot entries from the configuration and leave just one.'),
+            reporting.Flags([reporting.Flags.INHIBITOR])
+        ])
```

Comment:
@MichalHe  isn't this just a subcase of the problem that is this PR resolving? but I thought that in such a case we will see this & the other report in the same time always when `rescuse` entries are affected. Is it correct?